src/modules/member/services/member.service.ts

```diff
@@ -12,7 +15,75 @@ export class MemberService {
     private readonly prismaService: PrismaService,
     private readonly authService: AuthService,
   ) {}
+
+  /**
+   * member 단일 조회
+   */
+  findOne(where: Prisma.MemberWhereInput): Promise<MemberEntity> {
+    return this.prismaService.member.findFirst({
+      where,
+    }) as Promise<MemberEntity>;
```

Comment:
```suggestion
   findOne(where: Prisma.MemberWhereInput): Promise<MemberEntity | null> {
    return this.prismaService.member.findFirst({
      where,
    }) 
```

범용성 있는 사용을 위해 단일 조회를 findUnique가 아닌 findFirst로 선택 하신 것  좋은 선택이었다고 생각합니다.

그러나 findFirst와 같은 경우 where 조건에 해당하는 유저가 없다면 null을 return 해주는 것으로 알고 있는데, as 로 MemberEntity 타입을 단정 짓는 것이 아닌 return 타입에 유니온으로 null을 명시해주는 건 어떤가요?

Reply:
타입 단언을 한 이유는 pipe 레이어에서 한번 검증을 하기때문인데 service 에서는 해당 정보에 상관없이 동작해야하니 예외처리를 하는게 맞을까요???

Reply:
> 타입 단언을 한 이유는 pipe 레이어에서 한번 검증을 하기때문인데 service 에서는 해당 정보에 상관없이 동작해야하니 예외처리를 하는게 맞을까요???

요것도 의견주세요

Reply:
범용성있게 findOne를 사용하는 경우 where 조건에 여러 다른 케이스들이 들어올 것으로 예상되는데,
그럴 때마다 앞단에서 pipe로 리소스 존재 유무를 확인하고 사용하는데 불편함이 있지 않을까요?

위의 적용하신 예시는 id를 통한 리소스 검증일 때여서 IdRequestParamDto와 IsRecord를 사용해 리소스 유무를 편하게 검증했지만 where 조건에 id가 아닌 값이 들어오는 경우에는 사용하기 어려울 것 같다는 생각이 듭니다.

Reply:
오 그러네요 상당히 인정하는 부분입니다.